### PR TITLE
Spark Backend Support for Gora (GORA-386)

### PR DESCRIPTION
**Pull request by kamaci**
Pull request of the work for GSoC project (Spark Backend Support for Gora) for Apache Gora.

### CODE REVIEW

File: gora-core/pom.xml [hunk 1]
```diff
@@ -141,6 +141,13 @@
       <artifactId>guava</artifactId>
     </dependency>
 
+    <!-- Spark dependency -->
```
Comment (lewismc) [inline gora-core/pom.xml]: Dependency definitions usually happen within parent pom.xml. Modules then inherit the dependency without versions. This also enables us best opportunity to synchronize dependencies across modules.

File: pom.xml [hunk 2]
```diff
@@ -893,6 +893,10 @@
             <groupId>ant</groupId>
             <artifactId>ant</artifactId>
           </exclusion>
+          <exclusion>
```
Comment (lewismc) [inline pom.xml]: Why is this required?

File: gora-tutorial/src/main/java/org/apache/gora/tutorial/log/LogAnalyticsSpark.java [hunk 3]
```diff
@@ -0,0 +1,200 @@
+/**
```
Comment (lewismc) [inline gora-tutorial/src/main/java/org/apache/gora/tutorial/log/LogAnalyticsSpark.java]: Can you please add an entry to https://github.com/apache/gora/blob/master/bin/gora which enables us to run this very easily from the command line? It will help with testing and validation of your code.

File: gora-tutorial/pom.xml [hunk 4]
```diff
@@ -108,6 +108,11 @@
       <artifactId>gora-cassandra</artifactId>
     </dependency>
 
+    <dependency>
```
Comment (lewismc) [inline gora-tutorial/pom.xml]: Thanks for adding this.

File: gora-hbase/src/test/java/org/apache/gora/hbase/mapreduce/TestHBaseStoreWordCount.java [hunk 5]
```diff
@@ -58,4 +59,11 @@ public void testWordCount() throws Exception {
     MapReduceTestUtils.testWordCount(cluster.getConf(), webPageStore, tokenStore);
   }
 
+  //todo fix config
+  @Ignore
```
Comment (lewismc) [inline gora-hbase/src/test/java/org/apache/gora/hbase/mapreduce/TestHBaseStoreWordCount.java]: I am OK for us ignoring the tests at this point in time. It is high desired though that we ensure the issue you've described at http://www.mail-archive.com/dev%40gora.apache.org/msg06020.html is addressed in a separate issue.

File: gora-mongodb/src/test/java/org/apache/gora/mongodb/mapreduce/TestMongoStoreWordCount.java [hunk 6]
```diff
@@ -55,4 +56,12 @@ public void testWordCount() throws Exception {
         webPageStore, tokenStore);
   }
 
+  //todo fix config
+  @Ignore
```
Comment (lewismc) [inline gora-mongodb/src/test/java/org/apache/gora/mongodb/mapreduce/TestMongoStoreWordCount.java]: I am OK for us ignoring the tests at this point in time. It is high desired though that we ensure the issue you've described at http://www.mail-archive.com/dev%40gora.apache.org/msg06020.html is addressed in a separate issue.

File: gora-core/pom.xml (hunk 1 above)
Comment (kamaci) [inline gora-core/pom.xml]: Spark version should be placed at parent pom and it's done.

File: pom.xml [hunk 7]
```diff
@@ -644,6 +644,7 @@
     <commons-io.version>1.3.2</commons-io.version>
     <restlet.version>2.3.1</restlet.version>
 
+    <spark.version>1.3.1</spark.version>
```
Comment (lewismc) [inline pom.xml]: Any reason to not use Spark version 1.4.1?

File: pom.xml (hunk 2 above)
Comment (kamaci) [inline pom.xml]: servlet dependency is excluded from spark, not ant. Without excluding it throws a java security exception. Motivation for it: https://issues.apache.org/jira/browse/SPARK-1693

File: gora-tutorial/src/main/java/org/apache/gora/tutorial/log/LogAnalyticsSpark.java (hunk 3 above)
Comment (kamaci) [inline gora-tutorial/src/main/java/org/apache/gora/tutorial/log/LogAnalyticsSpark.java]: LogAnalyticsSpark is let to run from command line by passing "loganalyticsspark" argument.

Comment (lewismc) [inline gora-tutorial/src/main/java/org/apache/gora/tutorial/log/LogAnalyticsSpark.java]: Acknowledged. We do already however add examples to the Gora script which I mentioned. I have this on my path and use it to test out some jobs based on new patches coming in. It is not a big deal but just makes your work easier to use. Up to you, if you do not wish to implement then don't. I will.

File: pom.xml (hunk 7 above)
Comment (kamaci) [inline pom.xml]: Spark version is changed to 1.4.1 from 1.3.1. I've tested both built-in tests of Gora and some specific tests for Spark.

File: gora-tutorial/src/main/java/org/apache/gora/tutorial/log/LogAnalyticsSpark.java (hunk 3 above)
Comment (kamaci) [inline gora-tutorial/src/main/java/org/apache/gora/tutorial/log/LogAnalyticsSpark.java]: I can implement them with pleasure if you have some more examples and want me add GoraSparkEngine versions of them?

Comment (lewismc) [inline gora-tutorial/src/main/java/org/apache/gora/tutorial/log/LogAnalyticsSpark.java]: We will work on a more comprehensive test suite within the 0.7 development drive.
